src/server.js

```diff
@@ -0,0 +1,150 @@
+          if (syntax_errors.length > 0) {
+            let message = {
+              type: 'subscription_fail',
+              errors: syntax_errors,
+              id: message_data.id,
+            };
+            connection.sendUTF(JSON.stringify(message));
+          } else {
+            const sub_id = message_data.id;
+            if (!connection.subscriptions) {
+              connection.subscriptions = {};
+            }
+            //set up trigger listeners
+            let msg_triggers = [];
+            if (message_data.triggers) {
+              msg_triggers = message_data.triggers;
+
+            } else {
+              if (this.options.triggerGenerator) {
+                msg_triggers = this.options.triggerGenerator(message_data);
+              }
+            }
+            msg_triggers.forEach((trigger) => {
+              let string_trigger = JSON.stringify(trigger);
+              if (! this.triggers[string_trigger]) {
+                this.triggers[string_trigger] = [{connection: connection, sub_id: sub_id}];
+              } else {
+                this.triggers[string_trigger].push({connection: connection, sub_id: sub_id});
+              }
+            });
+            //set up polling message
+            if (message_data.pollingInterval) {
+              let pollingId = setInterval(
+                () => {
+                  graphql.graphql(
+                    this.options.schema,
+                    message_data.query,
+                    this.options.rootValue,
+                    this.options.contextValue,
+                    message_data.variables,
+                    message_data.operationName
+                  ).then(function(response){
+                    let message = response;
+                    message.type = 'subscription_data';
+                    message.id = sub_id;
+                    connection.sendUTF(JSON.stringify(message));
+                  }, function(err) {
+                    let message = response;
+                    message.type = 'subscription_data';
+                    message.id = sub_id;
+                    connection.sendUTF(JSON.stringify(message));
+                  });
+                }, 
+                message_data.pollingInterval
+              );
+              message_data.pollingId = pollingId;
+            }
+
+            connection.subscriptions[sub_id] = message_data;
+          }
+          
+        } else if (message_data.type === 'subscription_end') {
+          const sub_id = message.data.id;
+          clearInterval(connection.subscriptions[sub_id][pollingId]);
+          delete connection.subscriptions[sub_id];
+        }
+      });
+
+      connection.on('close', function(reasonCode, description) {
+        console.log(" Peer " + connection.remoteAddress + ' disconnected.');
+      });
+    });
+  }
+
+  triggerAction(message_data) {
```

Comment:
I think messageData is a bit confusing as argument name here. maybe triggerObject if it's the same as before?